dhall-json/tasty/data/fromArbitraryJSON_12.0.0.dhall

```diff
@@ -0,0 +1,17 @@
+  λ(JSON : Type)
+→ λ ( json
+    : { array : List JSON → JSON
+      , bool : Bool → JSON
+      , null : JSON
+      , number : Double → JSON
+      , object : List { mapKey : Text, mapValue : JSON } → JSON
+      , string : Text → JSON
+      }
+    )
+→ json.object
+    [ { mapKey = "string", mapValue = json.string "ABC" }
+    , { mapKey = "null", mapValue = json.null }
+    , { mapKey = "array", mapValue = json.array ([] : List JSON) }
+    , { mapKey = "number", mapValue = json.number 1.0 }
+    , { mapKey = "bool", mapValue = json.bool False }
```

Comment:
Is this order of `mapKey`s defined by `json-to-dhall`?

I think ideally we'd preserve the order of keys in the JSON document. The next best thing would IMHO be to use ascending lexicographic ordering.

But that's something for another ticket.

Reply:
It's the order of keys in the `HashMap` stored inside the `Object` constructor of `Value`.  The `HashMap` does not preserve order (one of the deficiencies of the `aeson` package), so you don't get them in some hash-dependent order.

That said, I could include a fix in this pull request to sort them.

Reply:
> That said, I could include a fix in this pull request to sort them.

That seems reasonable to me, but I don't mind too much.